server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/VacationRoutesTest.java

```diff
@@ -0,0 +1,297 @@
+
+    private static final Domain DOMAIN = Domain.of("example.org");
+
+    private static final Vacation VACATION = Vacation.builder()
+        .enabled(false)
+        .fromDate(Optional.of(ZonedDateTime.parse("2021-09-13T10:00:00Z")))
+        .toDate(Optional.of(ZonedDateTime.parse("2021-09-20T19:00:00Z")))
+        .subject(Optional.of("I am on vacation"))
+        .textBody(Optional.of("I am on vacation, will be back soon."))
+        .htmlBody(Optional.of("<p>I am on vacation, will be back soon.</p>"))
+        .build();
+
+    private static String isoString(ZonedDateTime date) {
+        return date.format(DateTimeFormatter.ISO_DATE_TIME);
+    }
+
+    private WebAdminServer webAdminServer;
+    private MemoryVacationRepository vacationRepository;
+    private MemoryNotificationRegistry notificationRegistry;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        vacationRepository = new MemoryVacationRepository();
+        notificationRegistry = new MemoryNotificationRegistry(new DefaultZonedDateTimeProvider());
+
+        SimpleDomainList domainList = new SimpleDomainList();
+        domainList.addDomain(DOMAIN);
+        MemoryUsersRepository usersRepository = MemoryUsersRepository.withVirtualHosting(domainList);
+
+        VacationRoutes vacationRoutes = new VacationRoutes(
+            vacationRepository, notificationRegistry, usersRepository, new JsonTransformer());
+        this.webAdminServer = WebAdminUtils.createWebAdminServer(vacationRoutes).start();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer).build();
+        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+
+        usersRepository.addUser(Username.of(BOB), "secret");
+    }
+
+    @AfterEach
+    void tearDown() {
+        webAdminServer.destroy();
+    }
+
+    @Test
+    void getVacation() {
+        vacationRepository.modifyVacation(AccountId.fromString(BOB), VacationPatch.builderFrom(VACATION).build()).block();
+
+        when()
+            .get(VacationRoutes.VACATION + SEPARATOR + BOB)
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .contentType(ContentType.JSON)
+            .body("enabled", equalTo(VACATION.isEnabled()))
+            .body("fromDate", equalTo(isoString(VACATION.getFromDate().get())))
+            .body("toDate", equalTo(isoString(VACATION.getToDate().get())))
+            .body("subject", equalTo(VACATION.getSubject().get()))
+            .body("textBody", equalTo(VACATION.getTextBody().get()))
+            .body("htmlBody", equalTo(VACATION.getHtmlBody().get()));
+    }
+
+    @Test
+    void getVacationFails() {
+        vacationRepository.modifyVacation(AccountId.fromString(BOB), VacationPatch.builderFrom(VACATION).build()).block();
+
+        when()
+            .get(VacationRoutes.VACATION + SEPARATOR + CAROL)
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400);
```

Comment:
This fails because the user CAROL do not exist, correct?

Shouldn't it be a 404?

Also please assert the fields in the JSON body of the response (that should contain very useful information.)

Reply:
Agreed, 404 is more specific in case someone needs to analyze the result.